src/stream/src/executor/backfill/no_shuffle_backfill.rs

```diff
@@ -299,20 +310,27 @@ where
 
                                         break 'backfill_loop;
                                     }
-                                    Some(chunk) => {
-                                        // Raise the current position.
-                                        // As snapshot read streams are ordered by pk, so we can
-                                        // just use the last row to update `current_pos`.
-                                        current_pos =
-                                            Some(get_new_pos(&chunk, &pk_in_output_indices));
-
-                                        let chunk_cardinality = chunk.cardinality() as u64;
-                                        cur_barrier_snapshot_processed_rows += chunk_cardinality;
-                                        total_snapshot_processed_rows += chunk_cardinality;
-                                        yield Message::Chunk(mapping_chunk(
-                                            chunk,
-                                            &self.output_indices,
-                                        ));
+                                    Some(record) => {
+                                        // Buffer the snapshot read row.
+                                        if let Some(data_chunk) = builder.append_one_row(record) {
+                                            let ops = vec![Op::Insert; data_chunk.capacity()];
+                                            let chunk = StreamChunk::from_parts(ops, data_chunk);
+                                            // Raise the current position.
+                                            // As snapshot read streams are ordered by pk, so we can
+                                            // just use the last row to update `current_pos`.
+                                            current_pos =
+                                                Some(get_new_pos(&chunk, &pk_in_output_indices));
+
+                                            let chunk_cardinality = chunk.cardinality() as u64;
```

Comment:
Not a big deal but the indent is a bit too deep. Perhaps we can try to split the `execute_inner` function later.

Reply:
https://github.com/risingwavelabs/risingwave/pull/14544/files#r1454375129